plugins/storage/volume/default/src/main/java/org/apache/cloudstack/storage/datastore/lifecycle/CloudStackPrimaryDataStoreLifeCycleImpl.java

```diff
@@ -404,6 +406,12 @@ protected boolean createStoragePool(long hostId, StoragePool pool) {
         CreateStoragePoolCommand cmd = new CreateStoragePoolCommand(true, pool);
         final Answer answer = agentMgr.easySend(hostId, cmd);
         if (answer != null && answer.getResult()) {
+            HypervisorType hyperVisorType = getHypervisorType(hostId);
+            if (pool.getPoolType() == StoragePoolType.NetworkFilesystem && hyperVisorType == HypervisorType.VMware) {
+                GetStoragePoolCapabilitiesCommand capsCmd = new GetStoragePoolCapabilitiesCommand();
+                capsCmd.setPool(new StorageFilerTO(pool));
+                agentMgr.easySend(hostId, capsCmd);
```

Comment:
not checking for the answer and updating the results from it ?

Reply:
Fixed